native/src/components/Header.tsx

```diff
@@ -210,8 +210,12 @@ const Header = ({
 
     const previousParams = previousRoute.params
 
-    // Poi details are not opened in a new route
-    if (route.name === POIS_ROUTE) {
+    if (
+      // Poi details are not opened in a new route
+      route.name === POIS_ROUTE &&
+      // unless coming from a deep link, and then the previous route has no slug
+      !!(previousParams as { slug?: string }).slug
```

Comment:
You can just create two boolean variables with meaningful names to get rid of these comments.